src/dash-table/components/ControlledTable/index.tsx

```diff
@@ -163,6 +163,28 @@ export default class ControlledTable extends PureComponent<
         this.handleDropdown();
         this.adjustTooltipPosition();
 
+        const {active_cell} = this.props;
+
+        // Check if the focus is inside this table
+        if (this.containsActiveElement()) {
+            const active = this.getActiveCellAttributes();
+
+            // If there is an active cell and it does not have focus -> transfer focus
+            if (
+                active &&
+                active_cell &&
+                (active.column_id !== active_cell?.column_id ||
+                    active.row !== active_cell?.row)
+            ) {
+                const target = this.$el.querySelector(
+                    `td[data-dash-row="${active_cell.row}"][data-dash-column="${active_cell.column_id}"]:not(.phantom-cell)`
+                ) as HTMLElement;
+                if (target) {
+                    target.focus();
+                }
+            }
```

Comment:
Make sure that the active cell is focused if the active element is another data cell in the table (triggered by mouse click to change the selected_cells)